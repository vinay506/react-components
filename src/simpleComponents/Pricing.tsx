import React from 'react';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import PricePackCard from './PricePackCard/PricePackCard';
import styled from 'styled-components';
import reactIcon from '../images/react-icon.png';
import ContentOverCard from './ContentOverCard/ContentOverCard';
import ActivityCard from '../Dashboard/ActivityCard/ActivityCard';


function Pricing() {
    const clickMe = () => {
        console.log('clicked');
    }
    // <div style={{ width: '200px', height: '200px', border: '1px solid', position: 'relative' }}>
    //     <CustomBadge text={'Recommended'} position={'absolute'} top={'-10px'} bgColor={'grey'}></CustomBadge>
    // </div>

    // <h1>Pricing page</h1>
    //     <TextPrimary text='Packs to empower your preparation'></TextPrimary>
    //     <TextSecondary text='Packs & features'></TextSecondary>
    //     <div>
    //         <TextContent text='Compare between the features of packs, choose'></TextContent>
    //     </div>
    //     <TextBlur text="Current Pack"></TextBlur>
    //     <Link text='View Schedule' onLinkClick={clickMe} ></Link>
    //     <CustomButton type={BUTTON_TYPES.DENSE} text={'Get Started'} color={'blue'} bgColor={'white'} borderColor={'blue'} onButtonClick={clickMe}></CustomButton>
    const freePack = () => {
        console.log('free pack');
    }
    const secondPack = () => {
        console.log('second pack');
    }
    const recomendedPack = () => {
        console.log('recomonded pack');
    }

    const activityCardHandler = () => {
        console.log('Activity Card handler');
    }
    return (
        <PricingContainer>
            <br></br>
            <ActivityCard cardClick={activityCardHandler} logoImage={reactIcon} title="Tests" description="Take full, part or chapter tests to improve your performance.
"></ActivityCard>
            <br></br>

            <ContentOverCard>
                <div>
                    <TextPrimary text={'Packs to empower your preparation journey. '}></TextPrimary>
                    <TextPrimary text={'Cancel anytime with in 15 days.'}></TextPrimary>
                    <TextContent text={'Get Started with the pack best suited for you to prepare your upcoming events.'}></TextContent>
                </div>
            </ContentOverCard>
            <PricePackContainer>
                <PricePackCard titleText={'Free'}
                    packText={'Current pack'}
                    rupees={'0'} buttonText={'Get Started'}
                    buttonCallBack={freePack}></PricePackCard>
                <PricePackCard logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'}
                    buttonCallBack={secondPack}></PricePackCard>
                <PricePackCard badgeText={'Recommended'} logo={reactIcon} titleText={'JEE Comprehensive 1-Year Pack (for Class 11)'}
                    packText={'1 year pack Valid till June 2021'} targetText={'Target Year 2020'} scheduleLink={'View Schedule'}
                    rupees={'3000'} expiryYearText={'for 1 year'} buttonText={'Get Started'} buttonCallBack={recomendedPack}></PricePackCard>
            </PricePackContainer>
        </PricingContainer >
    )
}

const PricePackContainer = styled.div`
margin: 10px;
display: flex;
justify-content: center;
`
const PricingContainer = styled.div`
    text-align:center
`


export default Pricing
