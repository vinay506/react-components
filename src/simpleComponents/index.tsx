import TextSecondary from './TextSecondary/TextSecondary';
import TextPrimary from './TextPrimary/TextPrimary';
import TextContent from './TextContent/TextContent';
import TextBlur from './TextBlur/TextBlur';
import Link from './Link/Link';
import CustomBadge from './CustomBadge/CustomBadge';
import CustomButton from './CustomButton/CustomButton';
import CustomDropdown from './CustomDropdown/CustomDropdown';
import SquareCard from './SquareCard/SquareCard';
import RectangleCard from './RectangleCard/RectangleCard';
import RoundedBox from './RoundedBox/RoundedBox';
import CommentIcon from './Icons/CommentIcon/CommentIcon';
import NameWithSubText from './NameWithSubText/NameWithSubText';
import CustomTableComponent from './CustomTableComponent/CustomTableComponent';
import InfoIcon from './Icons/InfoIcon/InfoIcon';
import DisplayWithFlex from './DisplayWithFlex/DisplayWithFlex';
import Tooltip from './Tooltip/Tooltip';
import TextSecondaryRegular from './TextSecondaryRegular/TextSecondaryRegular';
import DateDetails from './DateDetails/DateDetails';

export {
    TextSecondary, TextPrimary, TextContent, TextBlur, Link, CustomBadge,
    CustomButton, CustomDropdown, SquareCard, RectangleCard, RoundedBox, CommentIcon,
    NameWithSubText,CustomTableComponent,InfoIcon,Tooltip,DisplayWithFlex,TextSecondaryRegular,
    DateDetails};
