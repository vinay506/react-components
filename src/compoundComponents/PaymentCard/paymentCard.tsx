/**
 *  
 * Descirption:
 *  It Card of payments
 * 
 *  */ 


import React from 'react'
import {TextPrimary,TextSecondary,SquareCard,RoundedBox} from '../../simpleComponents';



const PaymentCard=(props:any)=> {
    return (
        <SquareCard style={{textAlign:'left'}}>
            <RoundedBox style={props.avatar.styles}></RoundedBox>
            <TextPrimary text={props.primaryText.text} styles={textPrimaryStyles}></TextPrimary>
            <TextSecondary text={props.secondaryText.text} styles={textSecondaryStyles}></TextSecondary>
        </SquareCard>
    )
}

const textPrimaryStyles = {
    marginBottom:'5%',
    fontSize:'14px'
}

const textSecondaryStyles = {
    marginBottom:'5%',
    fontSize:'12px'
}

export default PaymentCard;
