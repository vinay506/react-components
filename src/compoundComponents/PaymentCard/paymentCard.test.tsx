import React from 'react';
import { shallow } from 'enzyme';
import PaymentCard from './paymentCard';
import { RoundedBox, TextPrimary, TextSecondary } from '../../simpleComponents';
import renderer from 'react-test-renderer'

describe('<payment card>', function () {
    let wrapper:any;
    let props:any;
    beforeEach(() => {
        props = {
            avatar: {
                styles: {
                    height: '40px',
                    width: '40px',
                    marginBottom: '20px'
                }
            },
            primaryText: {
                text: 'Cancel anything with in 15 days...'
            },
            secondaryText: {
                text: `Change your mind? Don't worry, you can cancel your pack anytime with in 15 days and get pull refund*`
            }
        };
        wrapper = shallow(<PaymentCard {...props}></PaymentCard>);
    })


    // Snapshot test
    it('payment card snapshot', () => {
        const tree = renderer.create(<PaymentCard {...props}></PaymentCard>).toJSON()
        expect(tree).toMatchSnapshot()
    })



    it('renders the RoundedBox component', () => {
        expect(wrapper.find(RoundedBox).length).toBe(1);
    });
    it('renders the TextPrimary component', () => {
        expect(wrapper.find(TextPrimary).length).toBe(1);
    });
    it('renders the TextSecondary component', () => {
        expect(wrapper.find(TextSecondary).length).toBe(1);
    });

    it('renders the matching primary text', () => {
        const eleList = wrapper.find(TextPrimary)
        if (eleList) {
            const text = eleList.dive().text();
            expect(text).toEqual(props.primaryText.text);
        } else {
            expect(true).toBe(false)
        }
    });

    it('renders the matching secondary text', () => {
        const eleList = wrapper.find(TextSecondary)
        if (eleList) {
            const text = eleList.dive().text();
            expect(text).toEqual(props.secondaryText.text);
        } else {
            expect(true).toBe(false)
        }
    });
})