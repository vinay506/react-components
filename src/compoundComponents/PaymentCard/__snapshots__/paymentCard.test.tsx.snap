// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<payment card> payment card snapshot 1`] = `
<div
  className="sc-AxheI eMcboF"
  style={
    Object {
      "textAlign": "left",
    }
  }
>
  <div
    className="sc-AxmLO cCwvup"
    style={
      Object {
        "height": "40px",
        "marginBottom": "20px",
        "width": "40px",
      }
    }
  />
  <div
    className="sc-AxirZ eUWaBq"
    id="textPrimary"
    style={
      Object {
        "fontSize": "14px",
        "marginBottom": "5%",
      }
    }
  >
    Cancel anything with in 15 days...
  </div>
  <div
    className="sc-AxjAm hLZUJk"
    id="textSecondary"
    style={
      Object {
        "fontSize": "12px",
        "marginBottom": "5%",
      }
    }
  >
    Change your mind? Don't worry, you can cancel your pack anytime with in 15 days and get pull refund*
  </div>
</div>
`;
